src/main/java/seedu/address/model/appointment/AppointmentContainsDatePredicate.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.model.appointment;
+
+import static seedu.address.commons.util.DateUtil.DATE_FORMATTER;
+
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Appointment}'s start or end date matches the date given.
+ * Returns True if the date given is null and the person has an appointment.
+ */
+public class AppointmentContainsDatePredicate implements Predicate<Person> {
+    private final LocalDate date;
+
+    public AppointmentContainsDatePredicate(LocalDate date) {
+        this.date = date;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        if (person.getAppointment() == null) {
+            return false;
+        } else if (this.date == null) {
+            return true;
+        }
+
+        LocalDate startDate = person.getAppointment().getStart().toLocalDate();
+        LocalDate endDate = person.getAppointment().getEnd().toLocalDate();
+
+        return !this.date.isBefore(startDate) && !this.date.isAfter(endDate);
```

Comment:
Just a nit, I think may be good to separate them out as different boolean statements first, for enhance readability and understandability. You may not have to take my suggestion though, as the isBefore and isAfter is relatively intuitive.
i.e
```java//
boolean predicateDateAfterStartDate = !this.date.isBefore(startDate);
boolean predicateDateBeforeEndDate = !this.date.isAfter(endDate);

// returns true when predicate date lies between appointment startDate and endDate
return predicateDateAfterStartDate && predicateDateBeforeEndDate;


```